lib/range.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+const rangeParser = require('parse-numeric-range');
+const union = require('array-union');
+
+function parseRange(files) {
+	const any = new Set();
+	const ranges = {};
+	const filenames = [];
+
+	for (const file of files) {
+		const [actualFilename, rangeStr] = splitRangeStr(file);
+
+		if (any.has(actualFilename)) {
+			continue;
+		}
+
+		if (rangeStr === '') {
+			any.add(actualFilename);
+			ranges[actualFilename] = [];
+			continue;
+		}
+
+		const lines = rangeParser.parse(rangeStr);
```

Comment:
Because of how this implementation handles globs I needed to have the range part as a string, and of course, we need another function to parse that part.

Reply:
Right, so this will be a lot simpler without globs 😄